src/file.coffee

```diff
@@ -62,6 +62,22 @@ class File
       else
         false
 
+  # Public: Deletes the file on disk that corresponds to `::getPath()`.
+  #
+  # Returns a {Promise} that resolves once the file is deleted on disk. It
+  # resolves to a boolean value that is true if the file was deleted or false if
+  # it never existed.
+  delete: ->
+    @exists().then (isExistingFile) =>
+      return false unless isExistingFile
+
+      Q.Promise (resolve, reject) =>
+        fs.unlink @getPath(), (error) ->
+          if error && error.code != 'ENOENT'
```

Comment:
This is because I know `fs.unlink()` fails when the file is missing while `fs.remove()` succeeds when the file/folder is missing. I figure the unit tests would catch it if those assumptions stop being true.